src/components/Controls.tsx

```diff
@@ -18,16 +18,56 @@ export default function Controls() {
         onChange={(e) => dispatch(setShowStraightArrows(e.target.checked))}
       />
       <label htmlFor='straight-arrows'>Use straight arrows</label>
-      <div>
-        <button disabled={!canUndo} onClick={() => dispatch(undo())}>
-          undo
-        </button>
-        <button disabled={!canRedo} onClick={() => dispatch(redo())}>
-          redo
-        </button>
-        <button onClick={() => dispatch(reset())}>clear</button>
-      </div>
+      <UndoManagement />
       <Export />
+      {import.meta.env.DEV && <PersistedStateDebugging />}
+    </div>
+  )
+}
+
+function UndoManagement() {
+  const dispatch = useDispatch()
+  const { canUndo, canRedo } = useCanUndoRedo()
+
+  return (
+    <div>
+      <button disabled={!canUndo} onClick={() => dispatch(undo())}>
+        undo
+      </button>
+      <button disabled={!canRedo} onClick={() => dispatch(redo())}>
+        redo
+      </button>
+      <button onClick={() => dispatch(reset())}>clear</button>
+    </div>
+  )
+}
+
+function PersistedStateDebugging() {
```

Comment:
I noticed some odd behaviour which I suspected was local storage related when I was poking around a couple weeks ago before our club. Did you add that because they are happening a lot?

Reply:
no, it was because the Big Refactor has some localStorage migrations and I wanted an easy way to undo/compare them. do you remember what issues you saw? or better yet: can you try to reproduce them?

Reply:
No, not at all. Just a vague memory that something was odd. I'll make sure to note next time. Sorry.

Reply:
no worries 😄
I'm hoping it happened while I was still messing around with the code and by the time we did the club it was "finalized"

Reply:
Me too!